frontend/src/components/authentication/CreateAccountForm.jsx

```diff
@@ -0,0 +1,310 @@
+    const isValide = validateName(firstName);
+    if (isValide) {
+      this.setState({ isValidFirstName: true, firstNameClassStyle: "valid-field" });
+    } else {
+      this.setState({ isValidFirstName: false, firstNameClassStyle: "invalid-field" });
+    }
+  };
+
+  lastNameValidation = () => {
+    const lastName = document.getElementById("last-name").value;
+    const isValide = validateName(lastName);
+    if (isValide) {
+      this.setState({ isValidLastName: true, lastNameClassStyle: "valid-field" });
+    } else {
+      this.setState({ isValidLastName: false, lastNameClassStyle: "invalid-field" });
+    }
+  };
+
+  emailValidation = () => {
+    const email = document.getElementById("email").value;
+    const isValide = validateEmail(email);
+    if (isValide) {
+      this.setState({ isValidEmail: true, emailClassStyle: "valid-field" });
+    } else {
+      this.setState({ isValidEmail: false, emailClassStyle: "invalid-field" });
+    }
+  };
+
+  passwordValidation = () => {
+    this.setState({ isFirstPasswordLoad: false });
+    const password = document.getElementById("password").value;
+    const isValide = validatePassword(password);
+    const passwordConfirmation = document.getElementById("password-confirmation").value;
+    //Password valid?
+    if (isValide) {
+      this.setState({ isValidPassword: true, passwordClassStyle: "valid-field" });
+    } else {
+      this.setState({ isValidPassword: false, passwordClassStyle: "invalid-field" });
+    }
+    //Password confirmation is the same?
+    if (password === passwordConfirmation) {
+      this.setState({
+        isValidPasswordConfirmation: true,
+        passwordConfirmationClassStyle: "valid-field"
+      });
+    } else {
+      this.setState({
+        isValidPasswordConfirmation: false,
+        passwordConfirmationClassStyle: "invalid-field"
+      });
+    }
+  };
+
+  areAllFieldsValid = () => {
+    const {
+      isValidFirstName,
+      isValidLastName,
+      isValidEmail,
+      isValidPassword,
+      isValidPasswordConfirmation
+    } = this.state;
+
+    if (
+      isValidFirstName &&
+      isValidLastName &&
+      isValidEmail &&
+      isValidPassword &&
+      isValidPasswordConfirmation
+    ) {
+      //did not use state because of maximum update depth exeeded error
+      SUBMIT_BTN_DISABLED = false;
+    } else {
+      SUBMIT_BTN_DISABLED = true;
+    }
+  };
+
+  render() {
+    const {
+      isValidFirstName,
+      isValidLastName,
+      isValidEmail,
+      isValidPassword,
+      isFirstPasswordLoad,
+      firstNameClassStyle,
+      lastNameClassStyle,
+      emailClassStyle,
+      passwordClassStyle,
+      isValidPasswordConfirmation,
+      passwordConfirmationClassStyle
+    } = this.state;
+
+    //check if all fields are valid
+    this.areAllFieldsValid();
+    return (
+      <div>
+        <div>
+          <div style={styles.createAccountContent}>
+            <h3>{LOCALIZE.homePage.createAccount.content.title}</h3>
```

Comment:
See (comment Ref 1)

Reply:
Fixed in commit #https://github.com/code-for-canada/project-thundercat/pull/152/commits/6ce81fd2552bf6633dd7149f12e43f64df4e3db2